apisix/plugins/xml-json-conversion.lua

```diff
@@ -0,0 +1,164 @@
+    return 200, xml2lua.toXml(cjson.decode(table_data))
+end
+
+local _switch_anonymous = {
+    ["json"] = function(content_type, req_body, to)
+        if "application/json" ~= content_type then
+            return 400, {message = "Operation not supported"}
+        end
+
+        local data, err = core.json.decode(req_body)
+        if not data then
+            core.log.error("invalid request body: ", req_body, " err: ", err)
+            return 400, {error_msg = "invalid request body: " .. err,
+                         req_body = req_body}
+        end
+        if to == 'xml' then
+            return json2xml(req_body)
+        else
+            return 400, {message = "Operation not supported"}
+        end
+    end,
+    ["xml"] = function(content_type, req_body, to)
+        if "text/xml" ~= content_type then
+            return 400, {message = "Operation not supported"}
+        end
+        if to == 'json' then
+            return xml2json(req_body)
+        else
+            return 400, {message = "Operation not supported"}
+        end
+    end
+}
+
+function _M.access(conf, ctx)
+    local req_body, err = core.request.get_body()
+    if err or req_body == nil or req_body == '' then
+        core.log.error("failed to read request body: ", err)
+        core.response.exit(400, {error_msg = "invalid request body: " .. err})
+    end
+
+    local from = conf.from
+    local to = conf.to
+    if from == to then
+        return req_body
+    end
+
+    local content_type = core.request.headers()["Content-Type"]
+    local _f_anon = _switch_anonymous[from]
+    if _f_anon then
+        return _f_anon(content_type, req_body, to)
+    else
+        return 400, {message = "Operation not supported"}
+    end
+end
+
+local function get_json()
+    local args = core.request.get_uri_args()
+    if not args or not args.from or not args.to then
+        return core.response.exit(400)
+    end
+
+    local req_body, err = core.request.get_body()
+    if err or req_body == nil or req_body == '' then
+        core.log.error("failed to read request body: ", err)
+        core.response.exit(400, {error_msg = "invalid request body: " .. err})
+    end
+
+    local from = args.from
+    local to = args.to
+    if from == to then
+        return req_body
+    end
+
+    local content_type = core.request.headers()["Content-Type"]
+    local _f_anon = _switch_anonymous[from]
+    if _f_anon then
+        return _f_anon(content_type, req_body, to)
+    else
+        return 400, {message = "Operation not supported"}
+    end
+end
+
+function _M.api()
+    return {
+        {
+            methods = {"GET"},
+            uri = "/apisix/plugin/xml-json-conversion",
+            handler = get_json,
+        }
+    }
+end
+
+--function _M.control_api()
+--    return {
+--        {
+--            methods = {"GET"},
+--            uris ={"/v1/plugin/xml-json-conversion"},
+--            handler = get_json,
+--        }
+--    }
+--end
```

Comment:
just curious, what is this comments for?

Reply:
Just registered a control api for test, and do not create a route or service

Reply:
done, Already deleted